multicluster/controllers/multicluster/core/remote_common_area.go

```diff
@@ -0,0 +1,351 @@
+func NewRemoteCommonArea(clusterID common.ClusterID, clusterSetID common.ClusterSetID, url string, secret *v1.Secret,
+	scheme *runtime.Scheme, localCommonArea LocalCommonArea, remoteCommonAreaManager RemoteCommonAreaManager,
+	clusterSetNamespace string, configNamespace string) (common.CommonArea, error) {
+	klog.InfoS("Create a RemoteCommonArea", "Cluster", clusterID)
+
+	crtData, token, err := getSecretCACrtAndToken(secret)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create manager for the RemoteCommonArea
+	klog.InfoS("Connecting to RemoteCommonArea", "Cluster", clusterID, "url", url)
+	config, err := clientcmd.BuildConfigFromFlags(url, "")
+	if err != nil {
+		return nil, err
+	}
+	config.BearerToken = string(token)
+	config.CAData = crtData
+	mgr, err := ctrl.NewManager(config, ctrl.Options{
+		Scheme:             scheme,
+		MetricsBindAddress: "0",
+		Namespace:          clusterSetNamespace,
+	})
+	if err != nil {
+		klog.ErrorS(err, "error creating manager for RemoteCommonArea", "Cluster", clusterID)
+		return nil, err
+	}
+
+	remoteClient, e := client.New(config, client.Options{Scheme: scheme})
+	if e != nil {
+		return nil, e
+	}
+
+	remote := &remoteCommonArea{
+		Client:                  remoteClient,
+		ClusterManager:          mgr,
+		ClusterSetID:            clusterSetID,
+		ClusterID:               clusterID,
+		config:                  config,
+		scheme:                  scheme,
+		Namespace:               clusterSetNamespace,
+		connected:               false,
+		localCommonArea:         localCommonArea,
+		remoteCommonAreaManager: remoteCommonAreaManager,
+	}
+
+	remoteCommonAreaManager.AddRemoteCommonArea(remote)
+
+	return remote, nil
+}
+
+/**
+ * getSecretCACrtAndToken returns the access credentials from Secret.
+ */
+func getSecretCACrtAndToken(secretObj *v1.Secret) ([]byte, []byte, error) {
+	caData, found := secretObj.Data[v1.ServiceAccountRootCAKey]
+	if !found {
+		return nil, nil, fmt.Errorf("ca.crt data not found in Secret %v", secretObj.GetName())
+	}
+
+	token, found := secretObj.Data[v1.ServiceAccountTokenKey]
+	if !found {
+		return nil, nil, fmt.Errorf("token not found in Secret %v", secretObj.GetName())
+	}
+
+	return caData, token, nil
+}
+
+func (r *remoteCommonArea) SendMemberAnnounce() error {
+	memberAnnounceList := &multiclusterv1alpha1.MemberClusterAnnounceList{}
+	if err := r.List(context.TODO(), memberAnnounceList, client.InNamespace(r.GetNamespace())); err != nil {
+		return err
+	}
+	var localClusterMemberAnnounce multiclusterv1alpha1.MemberClusterAnnounce
+	localClusterMemberAnnounceExist := false
+	if len(memberAnnounceList.Items) != 0 {
+		for _, memberAnnounce := range memberAnnounceList.Items {
+			if memberAnnounce.ClusterID == string(r.remoteCommonAreaManager.GetLocalClusterID()) {
+				localClusterMemberAnnounceExist = true
+				localClusterMemberAnnounce = memberAnnounce
+				break
+			}
+		}
+	}
+	ctx, cancelFunc := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
Maybe add some comments here.

Have you guys discussed any other way to implement timeout?

Reply:
And why we need to specify a timeout here? Did we observe the default behavior does not fit here?

Reply:
Im not sure of the internal details of the Update whether it can take a long time. I thought it was safe to put a timeout. Ill try to find out what the default behavior is and if it can take unbounded time. Meanwhile is there a concern in keeping the timeout, even if it is not needed, it should not change functionality ?

Reply:
I do not see you add timeout for Get and List. Probably we can remove it. Can add back when we do see issues.

Reply:
okay sure